VivistaServer/RoleController.cs

```diff
@@ -0,0 +1,201 @@
+						}
+						catch (Exception ex)
+						{
+							await LoadRolesPage(context, "Something went wrong");
+						}
+					}
+					else
+					{
+						await LoadRolesPage(context, "You can't delete your own admin role.");
+					}
+				}
+				else
+				{
+					await LoadRolesPage(context, "User doesn't exist");
+				}
+			}
+			else
+			{
+				CommonController.Write404(context);
+			}
+		}
+
+		private static async Task LoadRolesPage(HttpContext context, string error = "", string success = "")
+		{
+			var roleid = GetRoleId("Admin");
+
+			SetHTMLContentType(context);
+			var adminTask = Task.Run(async () =>
+			{
+				await using var connection = Database.OpenNewConnection();
+				return await Database.QueryAsync<User>(connection,
+					"SELECT users.username as username FROM users INNER JOIN user_roles ON users.userid = user_roles.userid WHERE roleid = @roleid;", context, new { roleid });
+			});
+
+			//var admins = await adminTask;
+			adminTask.Wait();
+
+
+			var templateContext = new TemplateContext(new
+			{
+				admins = adminTask.Result.Select(a => new { a.username }).ToList(),
+				//admins = admins.Select(a => new { a.username }).ToList(),
+				error = error,
+				success = success
+			});
+
+			await context.Response.WriteAsync(await HTMLRenderer.Render(context, "Templates\\roles.liquid", templateContext));
+		}
+
+		public static int GetRoleId(string role)
+		{
+			int roleid = 0;
+			try
+			{
+				foreach (var r in allRoles)
+				{
+					if (r.name == role)
+					{
+						roleid = (int)r.id;
+					}
+				}
+			}
+			catch (Exception ex)
+			{
+				CommonController.LogDebug(ex.StackTrace);
+			}
+			Debug.Assert((roleid > 0), $"{role} doesn't exist");
+			return roleid;
+		}
+
+		public static async Task LoadRoles()
+		{
+			using var connection = Database.OpenNewConnection();
+			var roles = await connection.QueryAsync<Role>(@"SELECT id, name FROM roles;");
+			allRoles = (List<Role>)roles;
+
+#if DEBUG
+			CommonController.LogDebug("All roles:");
+			foreach (var role in allRoles)
+			{
+				CommonController.LogDebug($"id: {role.id} - name: {role.name}");
+			}
+#endif
+		}
+
+		public async static Task<bool> UserRoleExist(int userid, int roleid, NpgsqlConnection connection, HttpContext context)
+		{
+			try
+			{
+				int count = await Database.QuerySingleAsync<int>(connection, "select count(*) from user_roles where userid = @userid AND roleid = @roleid", context, new { userid, roleid });
+				return count > 0;
+			}
+			catch
+			{
+				return false;
+			}
+		}
+
+		public async static Task<bool> IsUserAdmin(HttpContext context)
+		{
+			var user = await UserSessions.GetLoggedInUser(context);
+			if (user != null)
+			{
+				using var connection = Database.OpenNewConnection();
+				var roles = await user.GetRoles(connection, context);
+				var adminId = RoleController.GetRoleId("Admin");
+				if (roles != null && roles.Contains(adminId))
+				{
+					return true;
+				}
```

Comment:
Kan je niet beter ook nog een functie voorzien waarmee je op eender welke Role kan checken?

Reply:
Bij voorkeur in User, denk ik.

Reply:
Dat kan ik ja, momenteel leek mij wel bij admin het beste om daar een aparte functie voor te hebben, omdat dit misschien nog vaker gaat terug komen.

Reply:
Wat bedoel je met voorkeur in User?

Reply:
Admin mag nog steeds een aparte functie hebben. Maar voorzie ook al een functie die op een welke Role kan checken. Dan kan je die functie hier ook aanroepen.

De functie die check op een role staat denk ik best in User.

Reply:
Oke dan zal ik die functie aanmaken en ze alle twee in de user klasse zetten